wrangler-core/src/main/java/io/cdap/wrangler/parser/GrammarBasedParser.java

```diff
@@ -93,12 +94,10 @@ public List<Directive> parse()
       });
 
       return result;
-    } catch (CompileException e) {
-      throw new DirectiveParseException(e.getMessage(), e.iterator(), e);
     } catch (DirectiveLoadException | DirectiveNotFoundException | DirectiveParseException e) {
-      throw e;
+      throw new RecipeException(e.getMessage(), e, directiveIndex.get());
     } catch (Exception e) {
-      throw new DirectiveParseException(e.getMessage(), e);
+      throw new RecipeException(e.getMessage(), e);
```

Comment:
Directive index needs to be added here.

Reply:
Not necessary because the other exceptions are thrown from functions where the directive index is not available. Will check again and update if needed later